clients/fides-js/src/fides-ext-gpp.ts

```diff
@@ -41,39 +45,87 @@ declare global {
 }
 
 /**
- * Wrapper around setting a TC string on the CMP API object
+ * Wrapper around setting a TC string on the CMP API object.
+ * Returns whether or not the TC string was set.
  * @param event: FidesEvent
  * @param cmpApi: the CMP API model
  */
 const setTcString = (event: FidesEvent, cmpApi: CmpApi) => {
+  if (!window.Fides.options.tcfEnabled) {
+    return false;
+  }
   const tcString = extractTCStringForCmpApi(event);
-  if (tcString) {
-    // Workaround for bug in base library https://github.com/IABTechLab/iabgpp-es/issues/35
-    cmpApi.setFieldValueBySectionId(TCF_SECTION_ID, "CmpId", ETHYCA_CMP_ID);
-    cmpApi.setSectionStringById(TCF_SECTION_ID, tcString);
+  if (!tcString) {
+    return false;
+  }
+  // Workaround for bug in base library https://github.com/IABTechLab/iabgpp-es/issues/35
+  cmpApi.setFieldValueBySectionId(TcfEuV2.ID, "CmpId", ETHYCA_CMP_ID);
+  cmpApi.setSectionStringById(TcfEuV2.ID, tcString);
+  return true;
+};
+
+/** From our options, derive what APIs of GPP are applicable */
+const getSupportedApis = () => {
+  const supportedApis: string[] = [];
+  // QUESTION: or should this come from GPP config?
+  if (window.Fides.options.tcfEnabled) {
```

Comment:
the GPP config also has a `regions` param which can include the EU. do we need to consider that as well when we say that TCF is enabled here inside GPP?

Reply:
Im not sure if it will help make this decision, but if a customer has purchased GPP and TCF, then TCF would be enabled as part of GPP.